plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryConnectorTest.java

```diff
@@ -331,6 +338,57 @@ protected boolean isColumnNameRejected(Exception exception, String columnName, b
         return nullToEmpty(exception.getMessage()).matches(".*(Fields must contain only letters, numbers, and underscores, start with a letter or underscore, and be at most 300 characters long).*");
     }
 
+    @Test
+    public void testPartitionDateColumn()
```

Comment:
It seems there are some special values as well. From https://cloud.google.com/bigquery/docs/partitioned-tables#date_timestamp_partitioned_tables

```
__NULL__: Contains rows with NULL values in the partitioning column.
__UNPARTITIONED__: Contains rows where the value of the partitioning column is earlier than 1960-01-01 or later than 2159-12-31.
```

Similar for range partitions where NULLs go to `__NULL__` and values outside the range of the partitioning go to `__UNPARTITIONED__`

Can you add a test for these?

Reply:
Thanks, I was missing that. Introduced another partition type `INGESTION` and added `testIngestionTimePartitionedTableInvalidValue()`.